src/test/run-make-fulldeps/reproducible-build/Makefile

```diff
@@ -1,4 +1,11 @@
 -include ../tools.mk
+
+# ignore-musl
+#
+# diff:
+# -/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustcORhwur/libunwind-cfdf9e23c318976b.rlib: 6220632559893696134
+# +/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustcMaw0kI/libunwind-cfdf9e23c318976b.rlib: 6220632559893696134
```

Comment:
Do you know what would be causing this? Is the toolchain used in the image not deterministic?

Reply:
I couldn't find out the reason, I'll search if I have the dump.

Reply:
Unfortunately premeditate artifacts are removed when the test fails, I have only this: https://www.diffchecker.com/eUuEetM2
Can I somehow prevent their removal?

Reply:
Checked with `-Csave-temps` and created temp files are exactly the same.
Whole message:
```
--- /checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/linker-arguments1	2019-03-05 17:16:53.182804827 +0000
+++ /checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/linker-arguments2	2019-03-05 17:16:53.376805672 +0000
@@ -30,8 +30,8 @@
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/libpanic_unwind-c76c1ae4d431675a.rlib: 1985635799303215181
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/libbacktrace_sys-cdf6c573b6895d26.rlib: 7956125457625004126
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/librustc_demangle-a6da7aad090b7964.rlib: 15614543219506318120
-/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustcFl4DM2/libunwind-4c72c22ad94926f9.rlib: 6336012346212133668
-/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustcFl4DM2/liblibc-2569e6208ce21483.rlib: 5871939339325629776
+/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustc9rtyT1/libunwind-4c72c22ad94926f9.rlib: 6336012346212133668
+/checkout/obj/build/x86_64-unknown-linux-musl/test/run-make-fulldeps/reproducible-build/reproducible-build/rustc9rtyT1/liblibc-2569e6208ce21483.rlib: 5871939339325629776
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/liballoc-2836f900f43284ca.rlib: 2657586149306604795
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/librustc_std_workspace_core-4a4d6a92861a5131.rlib: 2448695470350212496
 /checkout/obj/build/x86_64-unknown-linux-musl/stage2/lib/rustlib/x86_64-unknown-linux-musl/lib/libcore-a257b7c250777e7e.rlib: 1379652929685719242
Makefile:13: recipe for target 'smoke' failed
```

Reply:
This looks like it's possibly a false negative in this test? I thinkthe problem here is that the compiler is generating a temporary directory and temporarily modifying rlibs, because the two rlibs have the same contents they're just in different paths. Those different paths are fooling this test into thinking it has different inputs.

This is specific to the musl target which has crt-static shenanigans which necessitate some tweaking of the rlibs just before linking.

In that sense I think it's fine to ignore this test, but could the comment be updated to indicate that it actually passes it's just flagged as a false negative?

(or if you're intrepid the test can also be fixed!)

Reply:
I left that comment to memorize was wrong when I was debugging it but I don't know Rust internals well enough to fix it.
I'll improve the comment.